### ISSUE TITLE
MongoError: auth fails / Unable to define multiple connections Sails v0.10.5

### ISSUE THREAD

**Issue by brianjd** [balderdashy/sails#2479]
I experienced something similar with 0.10.5 as issue #2123 . After inspecting the connectionString, I found the issue that the password was being prefixed with whitespace.
~~~
error: A hook (`orm`) failed to load!
error: MongoError: auth fails
~~~

While debugging to fully isolate the whitespace issue, I noticed some more odd behaviour.  I made it to build-orm.js from the actual sails-mongo adapter, but even there while inspecting sails.config.connections, I found that the first connection's password would be fine, but the subsequent ones would have the whitespace prefix. I was able to work around it by simply assembling my own connectionString.  But it was a quick fix, and doesn't address the root cause - which I'm still working on isolating.  

Lastly, I found that when I have multiple connections (remote/sails-mongo) defined, I received the following:

~~~
error: A hook (`orm`) failed to load!
error: Error: failed to connect to [localhost:27017]
~~~

I highlight the connections being remote because for a reason unknown it is defaulting to localhost.  Also, because if there are two connections but one of them simply disk, it doesn't happen, and the remote connection is still successful.  Lastly, not sure if related, but I found that specifying the default connection name in env/production/development  or models.js doesn't make a difference.   I'm not familiar enough with the codebase, and too tired at the moment to continue this investigation :-P

**Comment by CWyrtzen** [comment]
Is this still going on?

**Comment by sailsbot** [comment]
Thanks for posting, @brianjd.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
